linkerd/app/integration/src/lib.rs

```diff
@@ -330,3 +343,58 @@ impl<T: fmt::Debug, E: fmt::Debug> ResultWaitedExt for Result<T, Waited<E>> {
         }
     }
 }
+
+pub async fn cancelable<E: Send + 'static>(
+    drain: drain::Watch,
+    f: impl Future<Output = Result<(), E>> + Send + 'static,
+) -> Result<(), E> {
+    drain.watch(Cancelable::new(f), |f| f.cancel()).await
```

Comment:
I think you could eliminate the Cancelable boilerplate with this:

```suggestion
    tokio::select! {
      res = &mut f => { res }
      shutdown = drain.signal() => {
        trace!("canceling...");
        let res = shutdown.release_after(f);
        trace!("canceled");
        res
      }
    };
```

Reply:
And actually, if all you want to do is drop the inner future, it can be even simpler...

Reply:
hmm, yeah, I think so (we _do_ need the `release_after` part here) --- this code predates the latest `drain` additions on main.

Reply:
It looks like Cancelable just drops the inner future at the moment? Unless I'm misreading it.

Reply:
ah, yeah, it was the code in your suggestion that _i_ misread :P